applications/GeoMechanicsApplication/tests/cpp_tests/test_time_stepping.cpp

```diff
@@ -93,24 +120,26 @@ KRATOS_TEST_CASE_IN_SUITE(ProcessMemberFunctionsAllCalledOnce, KratosGeoMechanic
     TimeStepExecutor executor;
     auto converging_strategy = std::make_shared<DummyStrategyWrapper>(TimeStepEndState::ConvergenceState::converged);
     executor.SetSolverStrategy(converging_strategy);
-    ProcessSpy spy;
-    TimeStepExecutor::ProcessRefVec process_refs{spy};
-    executor.SetProcessReferences(process_refs);
+    auto spy = std::make_shared<ProcessSpy>();
+
+    std::vector<std::shared_ptr<Process>> processes{spy};
+    std::vector<std::weak_ptr<Process>> process_observables(processes.begin(), processes.end());
```

Comment:
I believe we can simplify this to:
```suggestion
    std::vector<std::weak_ptr<Process>> process_observables{spy};
```

Reply:
Apparently so.
Done.